test_app/tests/lib/utils/test_validation.py

```
@@ -708,3 +710,319 @@ def test_validate_port_string_edge_cases(self, port_string, expected, descriptio
+        with pytest.raises(ValidationError):
+            validate_free_text(value)
+
+    @pytest.mark.parametrize(
+        "value,description",
+        [
+            ("%3Cscript%3Ealert(1)%3C/script%3E", "percent-encoded script tag"),
+            ("&#60;script&#62;alert(1)&#60;/script&#62;", "HTML-entity script tag"),
+            ("&#36;(whoami)", "HTML-entity shell substitution"),
+            ("%24%7BPATH%7D", "percent-encoded variable expansion"),
+            ("%3Ciframe%20src%3Dx%3E", "percent-encoded iframe tag"),
+            ("＜ｓｃｒｉｐｔ＞alert(1)", "fullwidth chars that NFKC-fold to script tag"),
```

Ruff (0.16.1)

[warning] String contains ambiguous `＜` (FULLWIDTH LESS-THAN SIGN). Did you mean `<` (LESS-THAN SIGN)?

(RUF001)